<?xml version="1.0"?>
<launch>
  <arg name="output" default="screen"/>
  <arg name="respawn" default="false"/>
  <arg name="start_manager" default="false"/>
  <arg name="start_server" default="false"/>
  <arg name="nodelet_manager" default="uwds_nodelet_manager"/>
  <arg name="launch-prefix" default=""/>

  <arg name="name" default="simple_object_provider"/>
  <arg name="input" default="color_detection"/>
  <arg name="object_name" default="orange_cup"/>
  <arg name="object_class" default="Cup"/>
  <arg name="object_model" default="$(find laas_objects)/res/blend/orange_cup.blend"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="output_world" default="simple_object"/>

  <node name="$(arg nodelet_manager)"
        pkg="nodelet" type="nodelet"
        args="manager"
        respawn="$(arg respawn)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)"
        if="$(arg start_manager)"/>

  <node name="uwds_server"
        pkg="nodelet" type="nodelet"
        args="load uwds/UwdsServerNodelet $(arg nodelet_manager)"
        respawn="$(arg respawn)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)"
        if="$(arg start_server)"/>

  <node name="$(arg name)"
        pkg="nodelet" type="nodelet"
        args="load uwds_basic_clients/SimpleObjectProvider $(arg nodelet_manager)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)">
    <remap from="~input" to="color_detection"/>
    <rosparam subst_value="true">
      global_frame_id: $(arg global_frame_id)
      object_name: $(arg object_name)
      object_class: $(arg object_class)
      object_model: $(arg object_model)
      output_world: $(arg output_world)
      verbose : false
    </rosparam>
  </node>

</launch>
